app/controllers/venues_controller.rb

```diff
@@ -0,0 +1,43 @@
+class VenuesController < ApplicationController
+  # To-Do : Use Pundit for better authorization
+	before_action :authenticate_user!, except: [:index, :show]
+  before_action :find_venue,  only: [:show, :update, :destroy]
+
+	def index
+		@venues = Venue.all
+    #Implement searching and pagination using Pagy gem
+	end
```

Comment:
Possible performance enhancement -

```suggestion
	def index
           @venues = Venue.all
           #Implement searching and pagination using Pagy gem
	end
	...
	private
	
	def load_venues
          @venues = Rails.cache.fetch("venues") do
            Venue.all
           end
        end
    end
```

Venue.rb model
```ruby
     after_commit :clear_cache_for_venues

     ...
     private

       def clear_cache_for_venues
         Rails.cache.delete("venues")
       end
```